Address Safer CPP warnings in MIMETypeRegistry.cpp and LocalizedStrings.cpp

cdumez opened the pull request:
#### 3483a327d94e2fce880cc582ae8f9c272af3463c
<pre>
Address Safer CPP warnings in MIMETypeRegistry.cpp and LocalizedStrings.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=298760">https://bugs.webkit.org/show_bug.cgi?id=298760</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/PAL/pal/ThreadGlobalData.h:
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(PAL::TextCodecICU::~TextCodecICU):
(PAL::TextCodecICU::createICUConverter const):
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/bindings/js/JSExecState.h:
(WebCore::JSExecState::currentState):
(WebCore::JSExecState::setCurrentState):
* Source/WebCore/dom/EventTarget.cpp:
(WebCore::EventTarget::removeAllEventListeners):
* Source/WebCore/dom/Microtasks.cpp:
(WebCore::MicrotaskQueue::performMicrotaskCheckpoint):
* Source/WebCore/dom/QualifiedName.cpp:
(WebCore::QualifiedName::QualifiedName):
(WebCore::QualifiedName::QualifiedNameImpl::~QualifiedNameImpl):
* Source/WebCore/dom/WindowEventLoop.cpp:
(WebCore::WindowEventLoop::breakToAllowRenderingUpdate):
* Source/WebCore/dom/make-event-names.py:
* Source/WebCore/loader/cache/CachedResourceRequestInitiatorTypes.h:
(WebCore::cachedResourceRequestInitiatorTypes):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCoreBundleSingleton):
(WebCore::copyLocalizedString):
(WebCore::webCoreBundle): Deleted.
* Source/WebCore/platform/MIMETypeRegistry.cpp:
(WebCore::MIMETypeRegistry::createMIMETypeRegistryThreadGlobalData):
(WebCore::MIMETypeRegistry::isSupportedImageMIMETypeForEncoding):
* Source/WebCore/platform/ThreadGlobalData.cpp:
(WebCore::ThreadGlobalData::setWebCoreThreadData):
(PAL::threadGlobalDataSingleton):
(PAL::threadGlobalData): Deleted.
* Source/WebCore/platform/ThreadGlobalData.h:
(WebCore::threadGlobalDataSingleton):
(WebCore::threadGlobalData): Deleted.
* Source/WebCore/platform/ThreadTimers.cpp:
(WebCore::ThreadTimers::setSharedTimer):
* Source/WebCore/platform/Timer.cpp:
(WebCore::threadGlobalTimerHeap):
(WebCore::ThreadTimerHeapItem::ThreadTimerHeapItem):
(WebCore::TimerBase::setNextFireTime):
(WebCore::TimerBase::fireTimersInNestedEventLoop):
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::forCurrentThread):
(WebCore::FontCache::forCurrentThreadIfExists):
(WebCore::FontCache::forCurrentThreadIfNotDestroyed):
* Source/WebCore/platform/ios/wak/WebCoreThread.mm:
(RunWebThread):
(StartWebThread):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::~MockRealtimeVideoSource):
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::destroyWorkerGlobalScope):
* Source/WebCore/workers/WorkerRunLoop.cpp:
(WebCore::RunLoopSetup::RunLoopSetup):
(WebCore::RunLoopSetup::~RunLoopSetup):

Canonical link: <a href="https://commits.webkit.org/299945@main">https://commits.webkit.org/299945@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d57bec141000ec02fdbf339a8ed0f192fea76d11

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120838 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40532 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31188 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127246 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72914 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/be2abac8-7545-46a1-927a-6ae9e1f0d2f8) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122714 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41230 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49109 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91780 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61027 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d5eaa874-a807-4124-ac33-5b0da30651a3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123790 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32917 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108318 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72478 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/88e901c6-18ed-440c-9a31-a55375217888) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31946 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26421 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70840 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102412 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26599 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130106 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47759 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36270 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100399 "Found 10 new test failures: imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic-opacity.html imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vs-float-clearance-002.html imported/w3c/web-platform-tests/css/css-transforms/animation/transform-non-invertible-discrete-interpolation.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-on-top-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-exit.html imported/w3c/web-platform-tests/css/css-writing-modes/inline-block-alignment-005.xht imported/w3c/web-platform-tests/css/css-writing-modes/inline-table-alignment-005.xht imported/w3c/web-platform-tests/css/filter-effects/filters-drop-shadow-002.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48127 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104492 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100302 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25430 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45686 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23732 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44440 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47621 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53326 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47092 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50436 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48776 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/platform/MIMETypeRegistry.cpp:
```diff
@@ -477,10 +477,10 @@ std::unique_ptr<MIMETypeRegistryThreadGlobalData> MIMETypeRegistry::createMIMETy
     for (CFIndex i = 0; i < count; i++) {
-        CFStringRef supportedType = reinterpret_cast<CFStringRef>(CFArrayGetValueAtIndex(supportedTypes.get(), i));
-        if (!isSupportedImageType(supportedType))
+        RetainPtr supportedType = reinterpret_cast<CFStringRef>(CFArrayGetValueAtIndex(supportedTypes.get(), i));
```
Can we use `checked_cf_cast` instead?

cdumez committed:
Address Safer CPP warnings in MIMETypeRegistry.cpp and LocalizedStrings.cpp
https://bugs.webkit.org/show_bug.cgi?id=298760

Reviewed by Ryosuke Niwa.

* Source/WebCore/PAL/pal/ThreadGlobalData.h:
* Source/WebCore/PAL/pal/text/TextCodecICU.cpp:
(PAL::TextCodecICU::~TextCodecICU):
(PAL::TextCodecICU::createICUConverter const):
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/bindings/js/JSExecState.h:
(WebCore::JSExecState::currentState):
(WebCore::JSExecState::setCurrentState):
* Source/WebCore/dom/EventTarget.cpp:
(WebCore::EventTarget::removeAllEventListeners):
* Source/WebCore/dom/Microtasks.cpp:
(WebCore::MicrotaskQueue::performMicrotaskCheckpoint):
* Source/WebCore/dom/QualifiedName.cpp:
(WebCore::QualifiedName::QualifiedName):
(WebCore::QualifiedName::QualifiedNameImpl::~QualifiedNameImpl):
* Source/WebCore/dom/WindowEventLoop.cpp:
(WebCore::WindowEventLoop::breakToAllowRenderingUpdate):
* Source/WebCore/dom/make-event-names.py:
* Source/WebCore/loader/cache/CachedResourceRequestInitiatorTypes.h:
(WebCore::cachedResourceRequestInitiatorTypes):
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::webCoreBundleSingleton):
(WebCore::copyLocalizedString):
(WebCore::webCoreBundle): Deleted.
* Source/WebCore/platform/MIMETypeRegistry.cpp:
(WebCore::MIMETypeRegistry::createMIMETypeRegistryThreadGlobalData):
(WebCore::MIMETypeRegistry::isSupportedImageMIMETypeForEncoding):
* Source/WebCore/platform/ThreadGlobalData.cpp:
(WebCore::ThreadGlobalData::setWebCoreThreadData):
(PAL::threadGlobalDataSingleton):
(PAL::threadGlobalData): Deleted.
* Source/WebCore/platform/ThreadGlobalData.h:
(WebCore::threadGlobalDataSingleton):
(WebCore::threadGlobalData): Deleted.
* Source/WebCore/platform/ThreadTimers.cpp:
(WebCore::ThreadTimers::setSharedTimer):
* Source/WebCore/platform/Timer.cpp:
(WebCore::threadGlobalTimerHeap):
(WebCore::ThreadTimerHeapItem::ThreadTimerHeapItem):
(WebCore::TimerBase::setNextFireTime):
(WebCore::TimerBase::fireTimersInNestedEventLoop):
* Source/WebCore/platform/graphics/FontCache.cpp:
(WebCore::FontCache::forCurrentThread):
(WebCore::FontCache::forCurrentThreadIfExists):
(WebCore::FontCache::forCurrentThreadIfNotDestroyed):
* Source/WebCore/platform/ios/wak/WebCoreThread.mm:
(RunWebThread):
(StartWebThread):
* Source/WebCore/platform/mock/MockRealtimeVideoSource.cpp:
(WebCore::MockRealtimeVideoSource::~MockRealtimeVideoSource):
* Source/WebCore/workers/WorkerOrWorkletThread.cpp:
(WebCore::WorkerOrWorkletThread::destroyWorkerGlobalScope):
* Source/WebCore/workers/WorkerRunLoop.cpp:
(WebCore::RunLoopSetup::RunLoopSetup):
(WebCore::RunLoopSetup::~RunLoopSetup):

Canonical link: https://commits.webkit.org/299945@main
